src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java

```diff
@@ -323,37 +366,141 @@ public long getOffsetSafeToCommitToKafka() {
+                        + " Channel:{}, offsetToken:{}",
+                    this.getChannelName(),
+                    event.getResult()))
+        .build();
+  }
+
+  /**
+   * Returns the offset Token persisted into snowflake.
+   *
+   * <p>OffsetToken from Snowflake returns a String and we will convert it into long.
+   *
+   * <p>If it is not long parsable, we will throw {@link ConnectException}
+   *
+   * @return -1 if no offset is found in snowflake, else the long value of committedOffset in
+   *     snowflake.
+   */
+  private long fetchLatestCommittedOffsetFromSnowflake() {
     LOGGER.debug(
         "Fetching last committed offset for partition channel:{}",
         this.channel.getFullyQualifiedName());
-    String offsetToken = this.channel.getLatestCommittedOffsetToken();
-    if (offsetToken == null) {
-      LOGGER.info("OffsetToken not present for channelName:{}", this.getChannelName());
-      return NO_OFFSET_TOKEN_REGISTERED_IN_SNOWFLAKE;
-    } else {
-      try {
-        latestCommittedOffsetInSnowflake = Long.parseLong(offsetToken);
+    String offsetToken = null;
+    try {
+      offsetToken = this.channel.getLatestCommittedOffsetToken();
+      if (offsetToken == null) {
+        LOGGER.info("OffsetToken not present for channelName:{}", this.getChannelName());
+        return NO_OFFSET_TOKEN_REGISTERED_IN_SNOWFLAKE;
+      } else {
+        long latestCommittedOffsetInSnowflake = Long.parseLong(offsetToken);
         LOGGER.info(
             "Fetched offsetToken:{} for channelName:{}",
             latestCommittedOffsetInSnowflake,
-            this.getChannelName());
+            this.channel.getFullyQualifiedName());
         return latestCommittedOffsetInSnowflake;
-      } catch (NumberFormatException e) {
-        final String errorMsg =
-            String.format(
-                "The offsetToken string does not contain a parsable long: %s,"
-                    + " offsetTokenLastRetrieved: %s. ",
-                this.getChannelName(), offsetToken);
-        LOGGER.error(errorMsg);
-        throw new ConnectException(errorMsg, e);
       }
+    } catch (NumberFormatException ex) {
+      LOGGER.error(
+          "The offsetToken string does not contain a parsable long:{} for channel:{}",
+          offsetToken,
+          this.getChannelName());
+      throw new ConnectException(ex);
     }
   }
 
+  /* Open a channel for Table with given channel name and tableName */
+  private SnowflakeStreamingIngestChannel openChannelForTable() {
+    OpenChannelRequest channelRequest =
+        OpenChannelRequest.builder(channelName)
+            .setDBName(this.databaseName)
+            .setSchemaName(this.schemaName)
+            .setTableName(this.tableName)
+            .setOnErrorOption(OpenChannelRequest.OnErrorOption.CONTINUE)
+            .build();
+    LOGGER.info(
+        "Opening a channel with name:{} for table name:{}", this.channelName, this.tableName);
```

Comment:
nit: is it possible we distinguish between open and reopen for better debugging purpose?

Reply:
Yes.. I actually add more debugging statements from reopening the channel.